Jenkinsfile-Release

```diff
@@ -12,22 +12,57 @@
  */
 @Library(['private-pipeline-library', 'jenkins-shared', 'int-jenkins-shared']) _
 
-def jira = [
+final jira = [
     versionPrefix: '', project: 'INT', projectId: '12410',
     credentialId : 'jenkins-jira', autoRelease: true, failOnError: true
 ]
 
-def jiraVersionMappings = [
-    'nexus-iq-server':'helm-iq', 
-    'nexus-repository-manager':'helm-nxrm'
+final jiraVersionMappings = [
+  'nexus-iq-server': 'helm-iq',
+  'nexus-repository-manager': 'helm-nxrm',
 ]
 
+final chartLocation = [
+  'nexus-iq-server': 'nexus-iq',
+  'nexus-repository-manager': 'nexus-repository-manager',
+]
+
+properties([
+  parameters([
+    choice(
+      choices: ['', 'nexus-iq-server', 'nexus-repository-manager'],
+      name: 'chart',
+      description: 'Chart to deploy.',
+    ),
+    string(
+      name: 'chartVersion',
+      description: 'Version of the Chart, like "139.0.0".',
+    ),
+    string(
+      name: 'appVersion',
+      description: 'Version of the application image, like "1.139.0"',
+    ),
+  ])
+])
+
 dockerizedBuildPipeline(
   prepare: {
+    if (! params.chart) {
+      error('Chart parameter is required.')
+    }
+    if (! params.chartVersion || ! params.appVersion) {
+      error('Version parameters are required.')
+    }
     githubStatusUpdate('pending')
   },
   buildAndTest: {
-    sh './build.sh'
+    final branch = gitBranch(env)
+    runSafely 'git config user.email "sonatype-ci@sonatype.com"'
+    runSafely 'git config user.name "Sonatype CI"'
+    runSafely "git checkout ${branch}"
+    runSafely "./upgrade.sh charts/${chartLocation[params.chart]} ${params.chartVersion} ${params.appVersion}"
```

Comment:
this is new: before running the build, use those versions we're given to run the previously-existing upgrade script.